src/DGtal/geometry/curves/AlphaThickSegmentComputer.h

```diff
@@ -0,0 +1,708 @@
+ * 
+ * - Finally apply the segment recognition (here of maximal thickness 15)  by adding the sequence (forward) of contour points: 
+ *  @snippet examples/geometry/curves/exampleAlphaThickSegmentNoisy.cpp exampleAlphaThickSegementNoisInitAndReco 
+ *  @note The maximal thickness given in initialization is the same than
+ *  the width given by the method width() of the ParallelStrip primitive
+ *  (not the vertical/horizontal width of the convex hull).
+ *
+ *  
+ * - If you use a Board2D display, you can draw the resulting segment like other 2D objects:
+ *  @snippet examples/geometry/curves/exampleAlphaThickSegmentNoisy.cpp exampleAlphaThickSegementDisplay 
+ * 
+ * The complete example of segment recognition is given in exampleAlphaThickSegmentNoisy.cpp
+ *
+ * @note You can also construct the segment by using an input point iterator in initialisation (see @ref moduleAlphaThickSegmentReco for more details) 
+ *
+ * The proposed implementation is mainly a backport from
+ * [ImaGene](https://gforge.liris.cnrs.fr/projects/imagene) with some
+ * various refactoring.
+ */
+
+
+template <typename TInputPoint,
+          typename TConstIterator = typename std::vector< TInputPoint >::const_iterator >
+
+class AlphaThickSegmentComputer
+{
+
+  // ----------------------- public types --------------------------------------
+  BOOST_STATIC_ASSERT(( TInputPoint::dimension == 2 ));
+  BOOST_CONCEPT_ASSERT((boost_concepts::ReadableIterator<TConstIterator>)); 
+ 
+public:
+  /**
+   * Type of input point.
+   **/
+  typedef TInputPoint InputPoint;
+  
+  /**
+   * The container type of Input Point
+   **/
+  typedef std::vector< InputPoint > InputPointContainer;
+  typedef typename InputPointContainer::size_type Size;
+  typedef typename InputPointContainer::const_iterator ContainerConstIterator;
+  typedef typename DGtal::functions::Hull2D::ThicknessDefinition ThicknessDef;
+  
+  typedef typename InputPointContainer::iterator Iterator;
+  typedef TConstIterator ConstIterator;
+  typedef ParallelStrip< SpaceND< 2,  DGtal::int32_t > ,true,true> Primitive;  
+  
+  /**
+   * Type of embedded points 
+   * @see getBoundingBox, getBoundingBoxFromExtremPoints
+   */
+  typedef DGtal::PointVector<2, double> PointD; 
+
+  typedef AlphaThickSegmentComputer<InputPoint, ConstIterator> Self; 
+  typedef AlphaThickSegmentComputer<InputPoint, ReverseIterator<ConstIterator> > Reverse; 
+  typedef DGtal::InHalfPlaneBySimple3x3Matrix<InputPoint, typename InputPoint::Component> Functor;  
+  typedef typename  DGtal::MelkmanConvexHull<InputPoint, Functor>::ConstIterator ConvexhullConstIterator;
+  
+  // ----------------------- internal types --------------------------------------
+
+private: 
+  struct State{
+    DGtal::MelkmanConvexHull<InputPoint, Functor> melkmanCH;  
+    InputPoint lastFront; /** the last point added at the front of the alpha thick segment */
```

Comment:
either put the ```/**``` before the member, or consider ```/*!<``` to put the documentation after the member.*

http://www.stack.nl/~dimitri/doxygen/manual/docblocks.html